pandas/_libs/index_class_helper.pxi.in

```diff
@@ -57,7 +57,10 @@ cdef class {{name}}Engine(IndexEngine):
             with warnings.catch_warnings():
                 # e.g. if values is float64 and `val` is a str, suppress warning
                 warnings.filterwarnings("ignore", category=FutureWarning)
-                indexer = values == val
+                if val != val:
+                    indexer = values != values
```

Comment:
something like

```
{{ if dtype in ["float64", "float32"]}}
if val != val:
    indexer = np.isnan(values)
else:
    indexer = values == val
{{else}}
indexer = values == val
{{endif}}
```

Reply:
I did the same thing yesterday, this caused the test to fail. Weird...